[RFC] vim-patch:8.0.{1353,1389,1406,1420,1432,1500,1569,1634,1678,1727,1782,1805,1831}

mhinz opened the pull request:
__vim-patch:8.0.1353: QuickFixCmdPost is not used consistently__

    Problem:    QuickFixCmdPost is not used consistently.
    Solution:   Invoke QuickFixCmdPost consistently after QuickFixCmdPre.
                (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/1ed2276fd50f34e824eeae93d5a5ebfdf118be26

__vim-patch:8.0.1389: getqflist() items are missing if not set__

    Problem:    getqflist() items are missing if not set, that makes it more
                difficult to handle the values.
    Solution:   When a value is not available return zero or another invalid
                value. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/a6d4849c711379b773529afaed640455287ac934

__vim-patch:8.0.1406: difficult to track changes to a quickfix list__

    Problem:    Difficult to track changes to a quickfix list.
    Solution:   Add a "changedtick" value. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/b254af312d1696b12367085acfbe41a41b7f1ea5

__vim-patch:8.0.1420: accessing freed memory in vimgrep__

    Problem:    Accessing freed memory in vimgrep.
    Solution:   Check that the quickfix list is still valid. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/3c09722600e3218905b5d4a7b635a9e6560f87b3

__vim-patch:8.0.1432: after ":copen" can't get the window-ID of the quickfix window__

    Problem:    After ":copen" can't get the window-ID of the quickfix window.
            (FalacerSelene)
    Solution:   Make it work without a quickfix list.  Add a test. (Yegappan
            Lakshmanan)

https://github.com/vim/vim/commit/2ec364e94dbc080ccdf6c5dfc6f1653b5b7ded64

__vim-patch:8.0.1500: possible NULL pointer dereference__

    Problem:    Possible NULL pointer dereference. (Coverity)
    Solution:   Check for the pointer not being NULL.

https://github.com/vim/vim/commit/0549a1e184d33674f4c2b8fb44ccdf6b9b9808a3

__vim-patch:8.0.1569: warning for uninitialized variable from gcc__

    Problem:    Warning for uninitialized variable from gcc.
    Solution:   Initialize the variable.

https://github.com/vim/vim/commit/28ada699c13833219acaeb7e768acc2acae50e02

__vim-patch:8.0.1634: the ex_vimgrep() function is too long__

    Problem:    The ex_vimgrep() function is too long.
    Solution:   Split it in smaller functions. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/75b0a888e41bcda4163072f41bb7b5471fef7651

__vim-patch:8.0.1678: errorformat "%r" implies "%>"__

    Problem:    Errorformat "%r" implies "%>". (Jan Gosmann)
    Solution:   Jump to before setting fmt_ptr. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/e333e79f9bdff82432e0fd7fcf7ae30ef8e3d092

__vim-patch:8.0.1727: qf_get_properties() function is too long__

    Problem:    qf_get_properties() function is too long.
    Solution:   Refactor the code. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/353eeeaca269ed5e83900bd4a24dc6dc80bb4880

__vim-patch:8.0.1782: no simple way to label quickfix entries__

    Problem:    No simple way to label quickfix entries.
    Solution:   Add the "module" item, to be used instead of the file name for
                display purposes. (Martin Szamotulski, closes vim/vim#1757)

https://github.com/vim/vim/commit/d76ce852668635d81778cedacc2d3f021ed4e475

__vim-patch:8.0.1805: qf_parse_line() is too long__

    Problem:    qf_parse_line() is too long.
    Solution:   Split it in parts.  Properly handle vim_realloc() failing.
                (Yegappan Lakshmanan, closes vim/vim#2881)

https://github.com/vim/vim/commit/18cebf44177542e6658251bacf6152aa9009ca58

__vim-patch:8.0.1831: sometimes the quickfix title is incorrectly prefixed with ':'__

    Problem:    Sometimes the quickfix title is incorrectly prefixed with ':'.
    Solution:   Prepend the colon in another way. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/8b62e31003693fee4b288e7aea49170f032aeef3

mhinz committed:
vim-patch:8.0.1782: no simple way to label quickfix entries

Problem:    No simple way to label quickfix entries.
Solution:   Add the "module" item, to be used instead of the file name for
            display purposes. (Martin Szamotulski)

https://github.com/vim/vim/commit/d76ce852668635d81778cedacc2d3f021ed4e475

mhinz committed:
vim-patch:8.0.1805: qf_parse_line() is too long

Problem:    qf_parse_line() is too long.
Solution:   Split it in parts.  Properly handle vim_realloc() failing.
            (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/18cebf44177542e6658251bacf6152aa9009ca58

mhinz commented on src/nvim/quickfix.c:
```diff
@@ -1230,9 +1048,308 @@ static void qf_new_list(qf_info_T *qi, char_u *qf_title)
+      fields->errmsg = p;
+      fields->errmsglen = linelen + 1;
+    }
+    STRNCPY(fields->errmsg, linebuf, linelen);
```
I kept 5 occurrences of `STRNCPY`/`STRNCPY`, so the lint job is expected to fail.

mhinz committed:
vim-patch:8.0.1678: errorformat "%r" implies "%>"

Problem:    Errorformat "%r" implies "%>". (Jan Gosmann)
Solution:   Jump to before setting fmt_ptr. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/e333e79f9bdff82432e0fd7fcf7ae30ef8e3d092

justinmk commented:
looks like some "bool" should be "int":

```
../src/nvim/regexp.c:3307:21: runtime error: load of value 251, which is not a valid value for type 'bool'
    #0 0x14ff8c2 in bt_regexec_nl /home/travis/build/neovim/neovim/build/../src/nvim/regexp.c:3307:21
    #1 0x14f804e in vim_regexec_both /home/travis/build/neovim/neovim/build/../src/nvim/regexp.c:7244:16
```

mhinz commented:
@justinmk Sorry, that wasn't my latest local copy. I pushed it now.

I still have the problem of `test/functional/ex_cmds/quickfix_commands_spec.lua` failing, because it seems that the `text` item (e.g. `getqflist()[0].text`) doesn't get terminated properly and sometimes contains garbage characters or even text from an old buffer. Looking in that today.

mhinz committed:
vim-patch:8.0.1831: sometimes the quickfix title is incorrectly prefixed with ':'

Problem:    Sometimes the quickfix title is incorrectly prefixed with ':'.
Solution:   Prepend the colon in another way. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/8b62e31003693fee4b288e7aea49170f032aeef3

mhinz committed:
vim-patch:8.0.1634: the ex_vimgrep() function is too long

Problem:    The ex_vimgrep() function is too long.
Solution:   Split it in smaller functions. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/75b0a888e41bcda4163072f41bb7b5471fef7651

mhinz committed:
vim-patch:8.0.1420: accessing freed memory in vimgrep

Problem:    Accessing freed memory in vimgrep.
Solution:   Check that the quickfix list is still valid. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/3c09722600e3218905b5d4a7b635a9e6560f87b3

mhinz commented:
By now, I have fixed the tests and the ASAN issue (uninitalized bool in the Vim patch itself) locally, but a few other patches are needed as well. Heavily WIP at the moment. Reviewing now won't be useful.

yegappan commented:
>
> By now, I have fixed the tests and the ASAN issue (uninitalized bool in the
> Vim patch itself) locally, but a few other patches are needed as well.
>

Are you referring to initializing the rm_ic field in regmatch in the qf_parse_get_fields()
function to FALSE? This was fixed in patch 8.0.1831.

Regards,
Yegappan

mhinz commented:
@yegappan Yup, I "fixed" it by porting that exact patch. Thanks for your Vim contributions!

vheon commented on runtime/doc/eval.txt:
```diff
@@ -7017,6 +7017,8 @@ setqflist({list} [, {action}[, {what}]])		*setqflist()*
 				buffer
 		    filename	name of a file; only used when "bufnr" is not
 				present or it is invalid.
+		    module	name of a moudle; if given it will be used in
```
there is a typo here moudle -> module

mhinz committed:
vim-patch:8.0.1406: difficult to track changes to a quickfix list

Problem:    Difficult to track changes to a quickfix list.
Solution:   Add a "changedtick" value. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/b254af312d1696b12367085acfbe41a41b7f1ea5

mhinz committed:
vim-patch:8.0.1353: QuickFixCmdPost is not used consistently

Problem:    QuickFixCmdPost is not used consistently.
Solution:   Invoke QuickFixCmdPost consistently after QuickFixCmdPre.
            (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/1ed2276fd50f34e824eeae93d5a5ebfdf118be26

mhinz committed:
vim-patch:8.0.1389: getqflist() items are missing if not set

Problem:    getqflist() items are missing if not set, that makes it more
            difficult to handle the values.
Solution:   When a value is not available return zero or another invalid
            value. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/a6d4849c711379b773529afaed640455287ac934

mhinz committed:
tests: adjust to latest Vim patches

mhinz committed:
doc: update setqflist()

mhinz committed:
lint

mhinz commented:
The list of patches got a little bit longer.

The `lint` job is expected to fail. I didn't change 3 of its suggestions on purpose.

janlazo commented:
Missing vim-patch:8.0.1500  https://github.com/vim/vim/commit/0549a1e184d33674f4c2b8fb44ccdf6b9b9808a3

mhinz committed:
vim-patch:8.0.1500: possible NULL pointer dereference

Problem:    Possible NULL pointer dereference. (Coverity)
Solution:   Check for the pointer not being NULL.

https://github.com/vim/vim/commit/0549a1e184d33674f4c2b8fb44ccdf6b9b9808a3

mhinz committed:
vim-patch:8.0.1727: qf_get_properties() function is too long

Problem:    qf_get_properties() function is too long.
Solution:   Refactor the code. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/353eeeaca269ed5e83900bd4a24dc6dc80bb4880

mhinz committed:
vim-patch:8.0.1432: after ":copen" can't get the window-ID of the quickfix window

Problem:    After ":copen" can't get the window-ID of the quickfix window.
            (FalacerSelene)
Solution:   Make it work without a quickfix list.  Add a test. (Yegappan
            Lakshmanan)

https://github.com/vim/vim/commit/2ec364e94dbc080ccdf6c5dfc6f1653b5b7ded64

mhinz commented:
Everything but the lint job is green now. Have fun reviewing. ;-)

janlazo commented:
Can you add [8.0.1569](https://github.com/vim/vim/commit/28ada699c13833219acaeb7e768acc2acae50e02)?

justinmk commented on test/functional/viml/errorlist_spec.lua:
```diff
@@ -26,17 +26,11 @@ describe('setqflist()', function()
-    setqflist({}, 'r', '5')
-    command('copen')
-    eq(':5', get_cur_win_var('quickfix_title'))
-  end)
```
I guess this needs to be mentioned in following-head wiki.

mhinz replied:
Good catch. That used to be the case at one point, but then more Vim patches were added and now it works again.

I put the test back.

mhinz committed:
vim-patch:8.0.1569: warning for uninitialized variable from gcc

Problem:    Warning for uninitialized variable from gcc.
Solution:   Initialize the variable.

https://github.com/vim/vim/commit/28ada699c13833219acaeb7e768acc2acae50e02

justinmk commented:
@mhinz Thanks for this! Note that keeping whitespace/formatting in the test files is helpful for `patch` and also for comparing to upstream. E.g. I noticed that patch `8.0.1831` whitespace is different.
